### PR TITLE
📝 Update docstings on `networks.py`

### PR DESCRIPTION
**Pull request by Kludex**
CLOSES DOC-16

Selected Reviewer: @hramezani

### CODE REVIEW

**Comment by Kludex** [comment]
please review

File: pydantic/networks.py [hunk 1]
```diff
@@ -353,10 +382,12 @@ def _validate(cls, __input_value: NetworkType, _: core_schema.ValidationInfo) ->
...
     """Email address validation using https://pypi.org/project/email-validator/.
 
-    Notes:
-    * raw ip address (literal) domain parts are not allowed.
-    * "John Doe <local_part@domain.com>" style "pretty" email addresses are processed
-    * spaces are striped from the beginning and end of addresses but no error is raised
+    Note:
+        Note that:
+
+        * raw IP address (literal) domain parts are not allowed.
```
Comment (tpdorsey) [inline pydantic/networks.py:388]:
```suggestion
        * Raw IP address (literal) domain parts are not allowed.
```

File: pydantic/networks.py [hunk 2]
```diff
@@ -353,10 +382,12 @@ def _validate(cls, __input_value: NetworkType, _: core_schema.ValidationInfo) ->
...
+        * "John Doe <local_part@domain.com>" style "pretty" email addresses are processed
```
Comment (tpdorsey) [inline pydantic/networks.py:389]:
```suggestion
        * "John Doe <local_part@domain.com>" style "pretty" email addresses are processed.
```

File: pydantic/networks.py [hunk 3]
```diff
@@ -353,10 +382,12 @@ def _validate(cls, __input_value: NetworkType, _: core_schema.ValidationInfo) ->
...
+        * spaces are striped from the beginning and end of addresses but no error is raised
```
Comment (tpdorsey) [inline pydantic/networks.py:390]:
```suggestion
        * Spaces are striped from the beginning and end of addresses, but no error is raised.
```

File: pydantic/networks.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+"""The networks module contains types for common network related fields."""
```
Comment (tpdorsey) [inline pydantic/networks.py:1]:
```suggestion
"""The networks module contains types for common network-related fields."""
```

**Review by tpdorsey — APPROVED** [review]
Minor suggested edits. Looks good.

**Pull request merged**